### DISCUSSION TITLE
Send hard reset over serial

### DISCUSSION THREAD

**Discussion by matthgyver** [micropython/micropython#9272] [Using MicroPython] [answered]
Hi,

Is it possible to send hard reset over serial please ?
Soft reset with `echo -ne "\x03\x04\r" > /dev/ttyACM0` work well but I can't find the same thing for hard reset.

Thank's

**Comment by robert-hh** [comment, marked as answer]
You can use machine.reset(). Just send `"\rimport machine;machine.reset()\r"`. To interrupt a running code, you may have to send Ctrlc-C beforehand and maybe wait a little bit.

**Reply by matthgyver** [reply]
Thank's Robert. Your answer help me to understand why that I tried does'nt work.

I just tested that and it work well without an active serial connection:
```
echo -ne "\x03\r" > /dev/ttyACM0 && echo -ne "\rimport machine;machine.reset()\r" > /dev/ttyACM0
```

**Reply by jimmo** [reply]
@matthgyver FWIW rather than doing this manually, use pyboard.py as a library for interacting with the device. See https://docs.micropython.org/en/latest/reference/pyboard.py.html#using-the-pyboard-library
